lib/sober_swag/nodes/binary.rb

```diff
@@ -1,37 +1,43 @@
 module SoberSwag
   module Nodes
     ##
-    # A
-    #
-    # It's cool I promise.
+    # A binary node: has a left and right hand side.
+    # Basically a node of a binary tree.
     class Binary < Base
+      ##
+      # @param lhs [SoberSwag::Nodes::Base] the left-hand node.
+      # @param rhs [SoberSwag::Nodes::Base] the right-hand node.
       def initialize(lhs, rhs)
         @lhs = lhs
         @rhs = rhs
       end
 
-      attr_reader :lhs, :rhs
+      ##
+      # @return [SoberSwag::Nodes::Base] the left-hand node
+      attr_reader :lhs
 
       ##
-      # Map the root values of the node.
-      # This just calls map on the lhs and the rhs
-      def map(&block)
-        self.class.new(
-          lhs.map(&block),
-          rhs.map(&block)
-        )
-      end
+      # @return [SoberSwag::Nodes::Base] the right-hand node
+      attr_reader :rhs
 
+      ##
+      # Deconstructs into an array of `[lhs, rhs]`
+      #
+      # @return [Array(SoberSwag::Nodes::Base, SoberSwag::Nodes::Base)]
       def deconstruct
         [lhs, rhs]
       end
 
+      ##
+      # Deconstruct into a hash of attributes.
       def deconstruct_keys(_keys)
         { lhs: lhs, rhs: rhs }
       end
 
```

Comment:
Would it be useful to add `#is_left?` and `#is_right?` like in [this binary tree implementation](http://rubytree.anupamsg.me/rdoc/Tree/BinaryTreeNode.html#is_left_child%3F-instance_method)?

Reply:
This isn't a node in a binary tree, it's a binary node. So that information is not really relevant, as this node *might* have a parent node that's like a metadata node or something.